lib/python/qmk/keymap.py

```diff
@@ -320,7 +320,7 @@ def generate_c(keymap_json):
     return new_keymap
 
 
-def write_file(keymap_filename, keymap_content):
+def write_file(keymap_filename: Path, keymap_content: str) -> Path:
```

Comment:
This function is only used by `write_json` and that isnt used anywhere, it doesnt make sense to keep them (and add test coverage for it).